system/lib/libc/dynlink.c

```diff
@@ -484,41 +484,55 @@ static void dlopen_onerror(struct dso* dso, void* user_data) {
 }
 
 // Modified version of path_open from musl/ldso/dynlink.c
-static int path_find(const char *name, const char *s, char *buf, size_t buf_size) {
+static int path_find(const char *name, int ncandidates, const char **candidates, char *buf, size_t buf_size) {
```

Comment:
Can we leave this function unmodified and instead write a wrapper function around it?

Also, I'm not sure about the name candidates, these are two different paths I guess?

Maybe just call `path_find` twice rather than being overly general like this?

Reply:
How's it look to you now?